### PR TITLE
Solved issue #1413

### PR DESCRIPTION
**Pull request by Ngai-E**
1. changed the email input type from 'text' to 'email'
2. validated filled email entry input to check fo validity using javaScript regex pattern match before saving.

### CODE REVIEW

**Comment by muarachmann** [comment]
yes Elizabeth thank you so much. I have some few request though :)

**Comment by Ngai-E** [comment]
> yes Elizabeth thank you so much. I have some few request though :


> yes Elizabeth thank you so much. I have some few request though :)

okay what are the requests?

**Comment by Ngai-E** [comment]
> @Ngai-E Elizabeth, if you are happy with the 35 character field size (SHOULD be enough!) then this is a good clean simple PR, and seems to function as desired.

Thanks

**Comment by aethelwulffe** [comment]
Yeah dude, what are the requests?  What did I miss in the review specifically?  Should we have used a different database type for this as well?  I didn't look into that.

File: templates/pharmacies/general_edit.php [hunk 1]
```diff
@@ -83,10 +83,24 @@
...
+
+            if(regex.test(email)){
+                top.restoreSession();
+                document.pharmacy.submit();
+                //Z&H Removed redirection
+            }
+
+            else{
+                alert("please enter a valid email address");
+            }
```
Comment (muarachmann) [inline templates/pharmacies/general_edit.php]: Here we intend to communicate messages to the user using izi-Toast - http://izitoast.marcelodolce.com/ it is included already in the codebase see its usage here https://github.com/LibreHealthIO/lh-ehr/pull/1392/files#diff-ac491a14571b3e954dca9d6842deaeb2 and

Comment (muarachmann) [reply templates/pharmacies/general_edit.php]: Also remove some trailing spaces and you are good to go

**Comment by muarachmann** [comment]
> Yeah dude, what are the requests? What did I miss in the review specifically? Should we have used a different database type for this as well? I didn't look into that.

oh nothing so complicated its actually good. just an enhancement :)

**Comment by Ngai-E** [comment]
Ok I will take a look at it @muarachmann  thanks for the review.

**Comment by muarachmann** [comment]
Great just that and you are good to go

**Pull request closed without merging**